### ISSUE TITLE
Dependencies shouldn't have pinned requirements

### ISSUE THREAD

**Issue by AlexandreDecan** [quantumblacklabs/causalnex#24]
The dependencies specified in `setup.py` directly come from the `requirements.txt` where they are specified with a pinned requirement (e.g. `pandas == 0.24.0`). This is no good practice for a package except in some edge cases (e.g. a known incompatibility or a known bug, but even in that case, it is preferable to declare a "less-than" requirement than an equality). 

Due to these pinned versions, it is not easy to use `causalnex` in an existing environment (where, e.g., another version than `pandas 0.24.0` is needed). 

Could I suggest you to declare more permissive constraints? It is quite common in Python to only define a lower bound (e.g. `pandas>=0.24.0`) but this is a very optimistic way of specifying a dependency constraint, since it is unlikely for a package to remain fully compatible with *all* future updates of a dependency. A better option would be to use `~=` (i.e. "all updates supposedly compatible with given version"). 

Btw, for the specific case of pandas, however, I suggest to adopt updates from the `1.0.0` branch as well.

**Comment by mzjp2** [comment]
Hi @AlexandreDecan, thanks for raising this! I fully agree and we have an internal ticket open tracking this as we loosen requirements and test that everything still works. Pandas 1.0 support will _probably_ come separately to that, I'd imagine.

I didn't know about `~=`, that's cool. Is it essentially the same as `pandas>=0.24.0, <1.0` (for e.g)?

**Comment by AlexandreDecan** [comment]
Thanks for your answer! 
Pandas 1.0 shouldn't be much of an issue since most changes were already announced (and led to deprecation warnings) in previous version. I can help you on this if needed. 

For `~=`, yes, it merely corresponds to the `~>` compatibility requirement in Rubygems. More information here: 
https://www.python.org/dev/peps/pep-0440/#compatible-release

TL;DR: You can assume `~=V.N` is equal to `>= V.N, == V.*`.

**Comment by SteveLerQB** [comment]
This issue has been solved. Closing this
